common/src/main/java/dev/isxander/yacl3/gui/controllers/PopupControllerScreen.java

```diff
@@ -0,0 +1,80 @@
+package dev.isxander.yacl3.gui.controllers;
+
+import dev.isxander.yacl3.gui.OptionListWidget;
+import dev.isxander.yacl3.gui.YACLScreen;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.util.Mth;
+
+public class PopupControllerScreen extends Screen {
+    private final YACLScreen backgroundYaclScreen;
+    private final OptionListWidget optionListWidget;
+    private final ControllerPopupWidget controllerPopup;
+    public double initialScrollAmount;
+    public double scrollAmount;
+    public double smoothScrollAmount;
+    public int controllerY;
+    public int maxScroll;
+    public PopupControllerScreen(YACLScreen backgroundYaclScreen, OptionListWidget optionListWidget, ControllerPopupWidget controllerPopup) {
+        super(controllerPopup.popupTitle()); //Gets narrated by the narrator - switch to translatable string?
+        this.backgroundYaclScreen = backgroundYaclScreen;
+        this.optionListWidget = optionListWidget;
+        this.controllerPopup = controllerPopup;
+        this.scrollAmount = optionListWidget.getScrollAmount();
+        this.controllerY = optionListWidget.getActivePopupControllerY();
+        this.maxScroll = optionListWidget.getMaxScroll();
+        this.initialScrollAmount = scrollAmount;
+        smoothScrollAmount = scrollAmount;
+    }
+
+
+    @Override
+    protected void init() {
+        this.addRenderableWidget(this.controllerPopup);
+    }
+
+    @Override
+    public void resize(Minecraft minecraft, int width, int height) {
+        minecraft.setScreen(backgroundYaclScreen);
+    }
+
+    @Override
+    public void renderBackground(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        smoothScrollAmount = Mth.lerp(Minecraft.getInstance().getDeltaFrameTime() * 0.5, smoothScrollAmount, scrollAmount);
+        double colorPickerY = controllerPopup.getDimension().height() - smoothScrollAmount + initialScrollAmount + controllerY - 20;
```

Comment:
`popupControllerY` or similar

Reply:
Fixed technically because its been removed in favor of the controller setting the popup widget y.